### PR TITLE
[sanitizer_symbolizer] RenderContextual elements for symbolizer markup.

### PR DESCRIPTION
**Pull request by avillega**
This is part of a stack of PRs to add support for symbolizer
markup in linux.

Render contextual symbolizer markup elements. For Fuchsia it is not
necessary to emit any context given that Fuchsia's logging
infrastructure already handles emitting it when necessary.

For more information about contextual symbolizer markup elements:
https://llvm.org/docs/SymbolizerMarkupFormat.html#contextual-elements

### CODE REVIEW

File: compiler-rt/lib/sanitizer_common/sanitizer_symbolizer_markup.cpp [hunk 1]
```diff
@@ -64,4 +66,111 @@ const char *MarkupSymbolizerTool::Demangle(const char *name) {
...
+
+    buffer->Append("\n");
+    accessBuffer.clear();
+  }
+}
+
+void MarkupStackTracePrinter::RenderContext(InternalScopedString *buffer) {
+  // Keeps track of the modules that have been rendered.
+  static bool initialized = false;
+  static InternalMmapVectorNoCtor<RenderedModule> renderedModules;
```
[the remark this answers is not in the document]
Comment (avillega) [reply compiler-rt/lib/sanitizer_common/sanitizer_symbolizer_markup.cpp:141]: The idea is to keep track of the modules that have been rendered so we don't render them again. On module load modules can be moved around in memory so we have to account for that as well that is why I need a struct to hold on some specific information about a module. When dlopen is called the ListOfModules of the `Symbolzer` gets invalidated and rebuilt, a lot of the time the already loaded modules will not change its location in memory but we can not know that unless we compare the modules we already emitted vs the modules that the `Symbolizer` holds. In my initial implementations the `Render*` functions, these function were free functions and I used static to keep the state in between calls. Now that I've put these functions in a class it makes sense to make the RenderedModules a member of the class instead.

**Comment by avillega** [comment]
I've addressed the review comments so far, please re review and leave comments if any.

**Comment by avillega** [comment]
Ping 👋

**Comment by vitalybuka** [comment]
Make sure to click "re-request review"

**Review by vitalybuka — APPROVED** [review]

**Review by petrhosek — APPROVED** [review]

**Pull request merged**